clients/TypeScript/packages/client/test/ChainSync.test.ts

```diff
@@ -11,76 +11,117 @@ import {
 
 const connection = { port: 1338 }
 
+const stubHandlers = {
+  rollBackward: () => {},
+  rollForward: () => {}
+}
+
 describe('ChainSync', () => {
   it('returns the interaction context', async () => {
-    const client = await createChainSyncClient({ connection })
+    const client = await createChainSyncClient(stubHandlers, { connection })
     expect(client.context.connectionString).toBe('ws://localhost:1338')
     expect(client.context.socket.readyState).toBe(client.context.socket.OPEN)
     await client.shutdown()
   })
-  it('selects the tip as the intersection if no point provided', async () => {
-    const client = await createChainSyncClient({ connection })
-    if (client.initialIntersection.point === 'origin' || client.initialIntersection.tip === 'origin') {
+
+  describe('startSync', () => {
+    it('selects the tip as the intersection if no point provided', async () => {
+      const client = await createChainSyncClient(stubHandlers, { connection })
+      const intersection = await client.startSync()
+      if (intersection.point === 'origin' || intersection.tip === 'origin') {
```

Comment:
The second part of the assertion looks odd to me. I don't see why the node's tip would point to origin in this context?

Reply:
It's a guard to ensure the chain tip has progressed past the `origin` point. If for some reason this is the case, the current state would produce the same result as the following test case `intersects at the genesis if origin provided as point`.